crates/iota-indexer/src/main.rs

```diff
@@ -19,26 +58,180 @@ async fn main() -> Result<(), IndexerError> {
+                })
+                .parse::<u64>()
+                .expect("Invalid INGESTION_READER_TIMEOUT_SECS");
+            let data_limit = std::env::var("CHECKPOINT_PROCESSING_BATCH_DATA_LIMIT")
+                .unwrap_or(
+                    IngestionConfig::DEFAULT_CHECKPOINT_DOWNLOAD_QUEUE_SIZE_BYTES.to_string(),
+                )
+                .parse::<usize>()
+                .unwrap();
+
+            let snapshot_min_lag = std::env::var("OBJECTS_SNAPSHOT_MIN_CHECKPOINT_LAG")
+                .ok()
+                .and_then(|s| s.parse::<usize>().ok())
+                .unwrap_or(SnapshotLagConfig::DEFAULT_MIN_LAG);
+
+            let rpc_client_url_parsed = old_conf
+                .rpc_client_url
+                .parse()
+                .expect("RPC Client url should be valid");
+
+            let command = if old_conf.analytical_worker {
+                Command::AnalyticalWorker
+            } else if old_conf.reset_db {
+                Command::ResetDatabase { force: true }
+            } else if old_conf.rpc_server_worker {
+                Command::JsonRpcService(JsonRpcConfig {
+                    iota_names_options: old_conf.iota_names_options,
+                    rpc_address: SocketAddr::new(
+                        old_conf
+                            .rpc_server_url
+                            .as_str()
+                            .parse()
+                            .expect("RPC Server url should be valid"),
+                        old_conf.rpc_server_port,
+                    ),
+                    rpc_client_url: old_conf.rpc_client_url,
+                })
+            } else {
+                Command::Indexer {
+                    ingestion_config: IngestionConfig {
+                        sources: IngestionSources {
+                            data_ingestion_path: old_conf.data_ingestion_path,
+                            remote_store_url: old_conf.remote_store_url.map(|url| {
+                                url.parse().expect("Remote Store URL should be correct")
+                            }),
+                            rpc_client_url: Some(rpc_client_url_parsed),
+                        },
+                        checkpoint_download_queue_size: download_queue_size,
+                        checkpoint_download_timeout: ingestion_reader_timeout_secs,
+                        checkpoint_download_queue_size_bytes: data_limit,
+                    },
+                    snapshot_config: SnapshotLagConfig {
+                        snapshot_min_lag,
+                        sleep_duration: SnapshotLagConfig::DEFAULT_SLEEP_DURATION_SEC,
+                    },
+                    pruning_options: PruningOptions {
+                        epochs_to_keep: std::env::var("EPOCHS_TO_KEEP")
+                            .map(|s| s.parse::<u64>().ok())
+                            .unwrap_or_else(|_e| None),
+                    },
+                }
```

Comment:
In this case it should fail. Currently if users don't define the kind of worker it leads to a noop.